File: hibernate-query-language/src/main/java/io/github/robertomike/hefesto/utils/StringUtils.java
package io.github.robertomike.hefesto.utils;

public final class StringUtils {

    public static final String EMPTY = "";

    public static final String[] EMPTY_ARRAY = new String[0];


    public static String[] splitByDot(String value) {
        return split(value, "\\.");
    }

    public static String[] split(String value, String regExpression) {
        String result = value == null ? EMPTY : value.trim();
        return result.isEmpty() ? EMPTY_ARRAY : result.split(regExpression);
    }
}

File: hibernate-query-language/src/main/kotlin/io/github/robertomike/hefesto/utils/StringUtils.kt
package io.github.robertomike.hefesto.utils

object StringUtils {
    const val EMPTY = ""
    val EMPTY_ARRAY = arrayOf<String>()

    @JvmStatic
    fun splitByDot(value: String): Array<String> {
        return split(value, "\\.")
    }

    @JvmStatic
    fun split(value: String, regExpression: String): Array<String> {
        val result = value.trim()
        return if (result.isEmpty()) EMPTY_ARRAY else result.split(regExpression.toRegex()).toTypedArray()
    }
}
